pkg/destroy/nutanix/nutanix.go

```diff
@@ -0,0 +1,227 @@
+		V3Client:  v3Client,
+		Logger:    logger,
+	}, nil
+}
+
+// Run is the entrypoint to start the uninstall process.
+func (o *ClusterUninstaller) Run() (*installertypes.ClusterQuota, error) {
+	category := createExpectedCategory(o.InfraID)
+	bootISOImageName := nutanixtypes.BootISOImageName(o.InfraID)
+	rhcosImageName := nutanixtypes.RHCOSImageName(o.InfraID)
+	imagesToDelete := []string{
+		bootISOImageName,
+		rhcosImageName,
+	}
+	o.Logger.Infof("Starting deletion of Nutanix infrastructure for Openshift cluster %s", o.InfraID)
+	clientV3 := o.V3Client.V3
+	//Delete VMs
+	err := cleanupVMs(clientV3, category, o)
+	if err != nil {
+		return nil, err
+	}
+
+	//Delete image
+	err = cleanupImages(clientV3, category, o, imagesToDelete)
+	if err != nil {
+		return nil, err
+	}
+	//Delete category
+	err = cleanupCategory(clientV3, category, o)
+	if err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+func cleanupVMs(clientV3 nutanixclientv3.Service, ec *expectedCategory, o *ClusterUninstaller) error {
+	matchedVirtualMachineList := make([]*nutanixclientv3.VMIntentResource, 0)
+	allVMsRaw, err := clientV3.ListAllVM(emptyFilter)
+	if err != nil {
+		return err
+	}
+	allVMs := allVMsRaw.Entities
+	for _, v := range allVMs {
+		vmName := *v.Spec.Name
+		if strings.HasPrefix(vmName, o.InfraID) {
+			if hasCategoryAssigned(v.Metadata, ec, o.Logger) {
+				if v.Spec.Description != nil {
+					if *v.Spec.Description == createdByOpenshiftInstallerDesc {
+						matchedVirtualMachineList = append(matchedVirtualMachineList, v)
+					}
+				}
+			}
+		}
+	}
+
+	amountOfTBD := len(matchedVirtualMachineList)
+	if amountOfTBD == 0 {
+		o.Logger.Infof("No VMs found that require deletion for cluster %s", o.ClusterID)
+	} else {
+		logToBeDeletedVMs(matchedVirtualMachineList, o.Logger)
+		err := deleteVMs(clientV3, matchedVirtualMachineList, o.Logger)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func cleanupImages(clientV3 nutanixclientv3.Service, ec *expectedCategory, o *ClusterUninstaller, expectedImageNames []string) error {
+	var found bool
+	allImagesRaw, err := clientV3.ListAllImage(emptyFilter)
+	if err != nil {
+		return err
+	}
+	allImages := allImagesRaw.Entities
+	for _, expectedImageName := range expectedImageNames {
+		found = false
+		for _, i := range allImages {
+			imageName := *i.Spec.Name
+			imageUUID := *i.Metadata.UUID
+			if imageName == expectedImageName {
+				if hasCategoryAssigned(i.Metadata, ec, o.Logger) {
+					if i.Spec.Description != nil {
+						if *i.Spec.Description == createdByOpenshiftInstallerDesc {
+							found = true
+							o.Logger.Infof("Deleting image %s with UUID %s", imageName, imageUUID)
+							response, err := clientV3.DeleteImage(imageUUID)
+							if err != nil {
+								return err
+							}
+							err = nutanixtypes.WaitForTask(clientV3, response.Status.ExecutionContext.TaskUUID.(string))
+							if err != nil {
+								o.Logger.Warnf("error while deleting images: %s", err)
```

Comment:
Should this error propagate back out instead of being logged?  It seems like this could lead to images being leaked if the image deletion never completes.